### ISSUE TITLE
Threading issues with mongo server timeout

### ISSUE THREAD

**Issue by dylanmcreynolds** [bluesky/bluesky-widgets#25]
In the relatively short period of time when the databroker browers was in xicam.gui, I spent some time trying to make it so that searching a catalog for runs didn't block xicam's main thread, and gave some notice to the user in the status bar about what was going on, with a progress bar (what I did needed improvement, for sure.)  Now that it's back to using bluesky-widgets, I'm seeing some issues with the new bluesky-widgets code in xicam.

While waiting for a mongo server timeout to occur, I see two issues:
- [ ] xicam is locked up, which indicates to me that something is blocking the main thread
- [ ] when the timeout does occur and control comes back xicam, the catalog selection dropdown does not re-enable, making it hard to try a different catalog.

**Comment by danielballan** [comment]
I believe that #27 addresses the first two items. The third item is still in the Xi-CAM layer; the bluesky-widgets component does not provide any Open button.

Taking just a quick look at the changes when I refactored to introduce the bluesky-widgets dependency (https://github.com/Xi-CAM/Xi-cam/pull/14/files#diff-334e756e951ddde184edaa4b5ebad6deL50-R117) I don't think anything was substantially changed there. Are we sure this didn't lock up before?

**Comment by dylanmcreynolds** [comment]
Agreed... one and two are fixed by your PR, and I the third is a xicam thing.
